[EASI-2906] TRB optional collaboration groups

aterstriep opened the pull request:
# EASI-2906

## Changes and Description

- Updated schema to make collaboration group checkboxes optional
- Fixed bug in collab groups question - subfields were not being cleared if form was reloaded
- Added "no answer selected" text to review component if no collab groups are selected

<!-- Put a description here! -->

## How to test this change

1. Create TRB request and navigate to request form
2. Confirm that collaboration group checkboxes are optional on first step
3. Confirm that subfield inputs are cleared when parent checkbox is unchecked
    - Answer will not clear from form UI unless page is refreshed
4. Confirm "no answer selected" text displays in review step

<!--
    Add any steps or code to run in this section to help others run your code:

    ```sh
    echo "Code goes here"
    ```
--->

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)


## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

patrickseguraoddball reviewed (changes requested):
LG, only thing is when the other fields are empty, the check and submit reads `null`

![Screenshot 2023-05-16 at 3 44 35 PM](https://github.com/CMSgov/easi-app/assets/95709965/aebcf31a-7bef-4be5-b71f-4a908660b6f2)

patrickseguraoddball commented:
Oh also I just found this item in the docs as a should have.  `GRT/GRB collaboration y/n sub-question should be required`

If the GRT/BRT item is selected, then that radio is required

aterstriep commented:
@patrickseguraoddball PR is updated with changes:

1. Fixed null values on Check and Submit page
2. Made GRB collab consult requested question required
3. Updated "all fields required" text to "fields marked with an asterisk" (from [UX review doc](https://docs.google.com/document/d/1HSmSXe6Wg7emBg59s3qFn7lINZnComsbpyVBlEJHb3Y/edit#heading=h.e537d9q37j1j))

patrickseguraoddball reviewed (approved):
LG
